aten/src/ATen/native/cuda/block_reduce.cuh

```diff
@@ -41,7 +42,6 @@ __inline__ __device__ T BlockReduceSum(T val, T* shared) {
   const int lid = threadIdx.x % C10_WARP_SIZE;
   const int wid = threadIdx.x / C10_WARP_SIZE;
   val = WarpReduceSum(val);
-  __syncthreads();
```

Comment:
these syncthreads are not redundant, they prevent races when a few BlockReduces are called in a row.

Reply:
Got it! I add comments instead.